CoreFoundation/Base.subproj/CFUtilities.c

```diff
@@ -1675,7 +1675,7 @@ CFDictionaryRef __CFGetEnvironment() {
         extern char **environ;
         char **envp = environ;
 #elif TARGET_OS_LINUX
-#if !defined(environ) && !TARGET_OS_ANDROID
+#if !defined(environ) && !TARGET_OS_ANDROID && defined(__GLIBC_PREREQ)
```

Comment:
I don't think that we should use `__GLIBC_PREREQ` to determine this, but rather `__GLIBC__` which has been present since glibc 6 (`__GNU_LIBRARY__` was used prior to that).